<launch>
  <arg name="ip_address" default="192.168.153.1" />

  <node name="slamware_ros_sdk_server_node" pkg="slamware_ros_sdk" type="slamware_ros_sdk_server_node" output="screen">

  <param name = "ip_address"                value = "$(arg ip_address)"/>
  <param name = "angle_compensate"         value = "true"/>
  <param name = "fixed_odom_map_tf"         value = "true"/>

  <param name = "robot_frame"              value = "/base_link"/>
  <param name = "odom_frame"               value = "/odom"/>
  <param name = "laser_frame"              value = "/laser"/>
  <param name = "map_frame"                value = "/slamware_map"/>

  <param name = "robot_pose_pub_period"    value = "0.05"/>
  <param name = "scan_pub_period"          value = "0.1"/>
  <param name = "map_pub_period"           value = "0.2"/>
  <param name = "path_pub_period"          value = "0.05"/>

  <param name = "/cmd_vel"                 value = "/cmd_vel"/>
  <param name = "/move_base_simple/goal"   value = "/move_base_simple/goal"/>
  <param name = "scan"                     value = "scan"/>
  <param name = "odom"                     value = "odom"/>
  <param name = "map"                      value = "map"/>
  <param name = "map_metadata"             value = "map_metadata"/>
  <param name = "global_plan_path"         value = "global_plan_path"/>

  <!-- topic remap /-->
  <remap from = "/cmd_vel"                   to = "/cmd_vel"/>
  <remap from = "/move_base_simple/goal"     to = "/move_base_simple/goal"/>
  <remap from = "scan"                       to = "scan"/>
  <remap from = "odom"                       to = "odom"/>

  <remap from = "map"                        to = "slamware_map"/>
  <remap from = "map_metadata"               to = "map_metadata"/>
  <remap from = "global_plan_path"           to = "global_plan_path"/>
  </node>

  <!--node name="map2odom" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 /slamware_map /odom 100"/-->
  <node name="base2laser" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 /base_link /laser 100"/>



</launch>
